kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -149,7 +149,13 @@ class BatchJobSubmission(session: KyuubiBatchSessionImpl, batchRequest: BatchReq
   override def close(): Unit = {
     if (!isClosedOrCanceled) {
       if (builder != null) {
-        builder.close()
+        try {
+          builder.shutdown()
+          applicationManager.killApplication(builder.clusterManager(), batchId)
```

Comment:
> For yarn-cluster mode and without application wait completion, it will kill the running spark app.

Sorry, I mistakenly thought the close method was called when the job finished, so how about we add it in `KyuubiBatchSessionImpl#close`?

Reply:
I think we can keep it with original.

We can only kill the application when killApp option is enabled.

```
<h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; color: rgb(36, 41, 47); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">DELETE /batches/${batchId}</h3>

Name | Description | Type
-- | -- | --
killApp | Whether to kill the running app | Boolean, optional, false by defaults

<br class="Apple-interchange-newline">
```

Reply:
Thanks, I will revert this